lib/cocoapods/installer/analyzer.rb

```diff
@@ -155,7 +155,15 @@ def sources
           if all_dependencies_have_sources
             sources = dependency_sources
           elsif has_dependencies? && sources.empty? && plugin_sources.empty?
-            sources = ['https://github.com/CocoaPods/Specs.git']
+            raise Informative, <<-EOS.strip_heredoc
+            There are no sources specified in your Podfile. Either: 
+              
+              a.) Add the master repo to your Podfile, ex. `source "master"`
+              b.) Add a private private repo at the source of your choosing, ex. `source "https://github.com/artsy/Specs"`
+
+            Note: The master repo is no longer included by default.
+
+            EOS
```

Comment:
We shouldn't do this if it breaks every existing Podfile relying on the default sources